File: php/src/Model/AttachmentUpload.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A presigned upload target. Send `headers` exactly as given - they are signed.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class AttachmentUpload
{
    public function __construct(
        public readonly string $url,
        public readonly string $method,
        /** @var array<string, string> */
        public readonly array $headers,
        public readonly int $expiresAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            url: (string) ($data['url'] ?? ''),
            method: (string) ($data['method'] ?? 'PUT'),
            headers: $data['headers'] ?? [],
            expiresAt: (int) ($data['expiresAt'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/AttachmentUpload.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * A presigned upload target. Send {@code headers} exactly as given — they are signed.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class AttachmentUpload {
    private String url;
    private String method;
    private Map<String, String> headers;
    private long expiresAt;

    public String getUrl() { return url; }
    public String getMethod() { return method; }
    public Map<String, String> getHeaders() { return headers; }
    public long getExpiresAt() { return expiresAt; }
}
